structuredheadings/plugin.js

```diff
@@ -326,19 +326,30 @@
             func = this.clearAutonumberClassesForHeading;
           } else {
             func = this.setAutonumberClassesForHeading.bind(this, value);
+
+             // prep the html for the collapsed, not in a heading case
+            if (headings === null) {
+              // avoid an extra snapshot created by the matchHeading command
+              editor.fire("lockSnapshot", { dontUpdate: true });
+              editor.execCommand("matchHeading");
+              editor.fire("unlockSnapshot");
+              // put the new heading into the headings array
+              headings = [CKEDITOR.plugins.structuredheadings.getCurrentBlockFromPath(editor)];
+            }
           }
 
-          if (headings.length > 0) {
+          if (headings !== null && headings.length > 0) {
```

Comment:
> How could headings be null here too?

In the "Clear headings" case we don't convert paragraphs to headings first.  The Clear heading code probably needs to be separated out at some later point if we keep this behaviour, but I'm not fussing too much until we have a successful usability test


Good point for the conditional though, `if (headings)` is probably sufficient here.